### ISSUE TITLE
Just getting 401 errors

### ISSUE THREAD

**Issue by camnealie** [electrickiwi-nz/api-samples#7]
Hello - I'm trying to use the nodejs example, sandbox, with a test account.

When I input my provided Client ID and secret and npm start, the code runs fine and going to localhost:5080 redirects me to welcome-dev.electrickiwi.co.nz/. So far so good.

When I enter the credentials outlined in developer.electrickiwi.co.nz/api-documentation/ , the form submits and nothing happens, except my email address and password get cleared. Looking at the console log of thebrowser, I see I've been navigated to a url including "error_code=401" so... 

Few notes:

First, I'm not sure if I should be entering 

api_dev+1@ekay.co.nz
or
api_dev1@ekay.co.nz

The documentation says the email is api_dev+[1-50]@ekay.co.nz, but I assume that's meant to be interpereted as 49 emails to choose from. 

Second, I've registered with a callback URL of "https://havent.got.one.yet.co.nz/" for obvious reasons. I'm not sure if I was supposed to register with a callback url of localhost or not? I haven't got an end use case for this integration yet, I just wanted to see what's possible.

Third, the welcome-dev.electrickiwi.co.nz/oauth site loads with an error in the console - 'cannot read properties of undefined (reading 'fn'). Not sure if that has any bearing here.

My initial test use for this integration was just to have a regularly updated display of when your hour of power is. Would really appreciate some more work to flesh out the examples, even if it's just with this initial sign in process.

**Comment by electrickiwi-nz** [comment]
@camnealie - Thanks for reaching out ... it looks like the sandbox is down at the moment and the team are looking into why this is the case. I will get back to you when this is back up. 

Specific to your questions:

- You are correct there are 50 users available using the pattern: api_dev+1@ekay.co.nz or api_dev+10@ekay.co.nz or api_dev+36@ekay.co.nz ...
- As part of OAuth2 [authorization flow](https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow) for Authorization Code Grant the token needs to be passed back to a valid URL. For the API Samples we have included http://localhost:5080/callback as part of your client credentials. When you go live we will need to have a valid callback URL accessible from the web. 
- The welcome-dev.electrickiwi.co.nz/oauth URL is not meant to be used directly - it is part of the OAuth2 paths and the two sandbox URLs you should use are:
Authorization URL: https://welcome-dev.electrickiwi.co.nz/oauth/authorize
Access Token URL: https://welcome-dev.electrickiwi.co.nz/oauth/token

We look forward to seeing your project come together - we believe the samples are working as expected but the problems you are having are because the sandbox is throwing errors. Sorry about this. 

I will be in touch shortly once it is up and running again. 

Cheers, Matt

**Comment by electrickiwi-nz** [comment]
@camnealie - Sorry for the wait. The sandbox is now back up and running. You should be able to perform the functions of the sample projects now using the login and code supplied. 

Thanks, Matt

**Comment by camnealie** [comment]
Thanks for that. Managed to login using localhost as the callback (again
haven't touched the code except to pop in my creds), but when it comes to
the /callback, I'm getting the following:


Error: [object Object]
    at getAuthError
(/Users/camnealie/Downloads/api-samples-master/node/src/node_modules/client-oauth2/src/client-oauth2.js:122:15)
    at
/Users/camnealie/Downloads/api-samples-master/node/src/node_modules/client-oauth2/src/client-oauth2.js:276:21
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  body: {
    status: 1,
    error: {
      code: 2,
      title: 'Client not recognised',
      detail: 'Invalid client access'
    }
  },
  code: 'EAUTH'
}

**Comment by electrickiwi-nz** [comment]
@camnealie - thanks for the message - sorry this is not working out as easy as planned. 

Can you confirm that you have followed the instructions on the readme.md file for:
- Installation
- Configuration
- Running

Because when I do them locally I don't get the error that you are describing. 

I run the following installation from the `node/src` directory:

```
npm install
```

I configure the application by copying your client id and secret into the [following](https://github.com/electrickiwi-nz/api-samples/blob/master/node/src/index.js#L8)
```
const clientId = 'YOUR_CLIENT_ID';
const clientSecret = 'YOUR_CLIENT_SECRET';
```

and run the application

```
npm start
```

I then browse to [http://localhost:5080/](http://localhost:5080/) and the login screen shows, I enter the api_dev+3@ekay.co.nz and the password `Ekay123456!` and then it does the redirects and I get the session data shown on screen. 

I don't browse to the `/callback` URL directly as it will throw an error - this URL is used in the OAuth2 authorization_grant flow to pass the Bearer token back to the application. 

If you look at the `/` route you will see that once you have a `token` you can interact through the `axios` library with the API. Look at line 76 of the [index.js](https://github.com/electrickiwi-nz/api-samples/blob/master/node/src/index.js#L76) file to see how you can frame and call the consumption endpoint. 

Let me know how you get on ... happy to help. 

Cheers, Matt

**Comment by camnealie** [comment]
Yes I can confirm I've done all those steps - did it from scratch just now
again with a fresh download of the project from the repo.

(I'm running Node v16.9.1 and npm 8.11.0 before you ask)

The code falls over at line
41:  electrickiwiAuth.code.getToken(req.originalUrl).then(function (user)
{....

If I throw a .catch after the .then I can even catch the error, but it's
the same as what I've shown above:

    status: 1,
    error: {
      code: 2,
      title: 'Client not recognised',
      detail: 'Invalid client access'
    }

I've created a loom video to demonstrate:
https://www.loom.com/share/68ff64cc85504f9b8b4e27e0f76da14d

**Comment by electrickiwi-nz** [comment]
@camnealie - thanks for the video and confirming the steps. I have retested with your specific client_id and client_secret and it is working as expected. I have also looked to the logs and see the `Invalid client` responses coming through but nothing about the client_id and client_secret. 

I am going to resend these back to you via email, incase there is an issue with either of them. 

Sorry for the hassle. Cheers, Matt
